### PR TITLE
External apps authentication

### PR DESCRIPTION
**Pull request by superalex**

### CODE REVIEW

**Comment by superalex** [comment]
Related to: https://github.com/taigaio/taiga-back/pull/447

File: app/modules/external-apps/external-app.controller.coffee [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    _getApplicationToken: =>
+        return @externalAppsService.getApplicationToken(@._applicationId, @._state)
+            .then (data) =>
+                @._application = data.get("application")
+                if data.get("auth_code")
+                    @._redirect(data)
+                    @._applicationTokenExists = true
+                else
+                    console.log 11111111
```
Comment (bameda) [inline app/modules/external-apps/external-app.controller.coffee]: :alien:

File: app/modules/external-apps/external-app.controller.coffee [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                    @._applicationTokenExists = false
+            .catch (xhr) =>
+                return @xhrError.response(xhr)
+
+    cancel: () ->
+        window.history.back()
+        $('head').find("meta[name='viewport']").remove()
```
Comment (bameda) [inline app/modules/external-apps/external-app.controller.coffee]: I move this line to a method _removeViewport and rename _changeViewport to _addViewport

File: app/modules/external-apps/external-app.jade [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        .card-inner
+            div.app-image
+                img(ng-src="{{::vm.application.get('icon_url')}}", alt="{{vm.application.get('name')}}")
+            div.app-data
+                h3 {{ ::vm.application.get("name") }}
+                a(ng-href="{{vm.application.get('web')}}", target="_blank") {{ ::vm.application.get('web') }}
+                p {{ ::vm.application.get("description") }}
+
+    a.button-green(href="#", ng-click="vm.createApplicationToken()", translate="EXTERNAL_APP.AUTHORIZE_APP")
+    a.cancel(href="#", ng-click="vm.cancel()", translate="EXTERNAL_APP.CANCEL")
```
Comment (bameda) [inline app/modules/external-apps/external-app.jade]: All links without title

File: app/modules/external-apps/external-app.controller.coffee (hunk 2 above)
Comment (superalex) [inline app/modules/external-apps/external-app.controller.coffee]: I added the methods addMobileViewport and removeMobileViewport to app/modules/services/app-meta.service.coffee and they are called properly when necesary from app.coffee but forgot removing the call from cancel.

**Comment by superalex** [comment]
@bameda I've updated the PR with all your comments, thanks! :+1:

**Comment by juanfran** [comment]
add more one-time-bindings in external-app.jade

**Comment by juanfran** [comment]
I would prefer to store disableHeader in a service instead of in the $rootScope

**Comment by juanfran** [comment]
All the pages that have "include ../includes/components/beta" don't have header so, I think you can remove the no-header class and change directly the `top` in beta.scss

**Comment by superalex** [comment]
Updated! ;)

**Comment by juanfran** [comment]
why do you need applicationTokenExists?

**Comment by superalex** [comment]
The old version of the template needed it, removed now.

**Comment by bameda** [comment]
Add this to CHANGELOG

**Comment by superalex** [comment]
CHANGELOG updated in both PR's (back and front)

**Comment by juanfran** [comment]
remove `div(class="no-header")` from login.jade

**Comment by superalex** [comment]
@juanfran it's fixed right now! :)

**Pull request merged**